core/issuers.chainmail

```diff
@@ -1,56 +1,174 @@
+  makeAmount(quantity:Quantity) -> (Amount);
+
+  /** Make an empty purse associated for this kind of right. */
+  makeEmptyPurse(name:String) -> (Purse);
+
+  /**
+   * Make a new Payment that has exclusive rights to the amount, only if the
+   * Purse had those rights, exclusively or not.
+   */
+  getExclusive(amount:Amount, src:Purse, name:String = 'a payment') -> (Payment);
+
+  /**
+   * Make a new Payment that has exclusive rights to all the contents of the
+   * purse.
+   */
+  getExclusiveAll(src:Purse, name:String = 'a payment') -> (Payment);
+
+  /**
+   * Burn (destroy) amount of the rights in src. If the rights are fungible,
+   * this directly removes the rights from the purse, while if the rights are
+   * non-fungible, this ensures that no purse has access to those rights. This
+   * step is often a prelude to re-issuing the rights.
+   */
+  burn(amount:Amount, src:Payment);
+
+  /** Burn (destroy) all of the rights in src. */
+  burnAll(src:Payment);
+}
+
+/**
+ * Makes Mints.
+ * 
+ * The description becomes part of the label, used by the assay to identify
+ * amounts authorized/acknowledged by the mint. The MintController and Assay
+ * must be compatible with the type of asset managed by the mint.
+ * 
+ * Description doesn't have to be a string, but it will only be used for its
+ * value.
+ */
+interface MintMaker {
+  makeMint(
+      description:String,
+      makeMintController:MintControllerMaker,
+	  makeAssay:AssayMaker)
+	-> (Mint);
+}
+
+/**
+ * Holding a Mint gives the exclusive right to control issuance and destruction
+ * of Purses containing amounts of a particular currency. Purses (and payments)
+ * associated with the issuer will only interact with others using the same
+ * mint.
+ */
+interface Mint {
+  /** Get the Issuer for this mint. */
+  getIssuer() -> (Issuer);
+
+  /**
+   * Create a new Purse containing the amount. Give it the specified name or "a
+   * purse".
+   */
+  mint(initialBalance:Amount, name:String = 'a purse') -> (Purse);
+
+  /** Reset all purses to contain nothing. */
+  destroyAll();
+
+  /** (Non-fungible amounts only.) Ensure no purse contains amount. */
+  destroy(amount:Amount);
+
+  /**
+   * Destroy any purses or payments containing amount, then issue a new Purse
+   * with that amount.
+   */
+  revoke(amount:Amount) -> (Purse);
+}
+
+/**
+ * Purses hold verified amounts of certain rights issued by Mints. Purses can
+ * transfer part of the balance they hold in a Payment, which has a narrower
+ * interface. A purse's balance can rise and fall, through the action of deposit
+ * and withdraw, while payments don't accept deposits.
+ *
+ * The primary use for Purses and Payments is for currency-like and goods-like
+ * valuables, but they can also be used to represent other kinds of rights, such
+ * as the right to participate in a particular contract.
+ *
+ * Currency-like assets are fungible, and don't distinguish use rights from
```

Comment:
This will have to be updated when we get your changes in, but I want to keep the description of currency-like assets that are used by spending them to see if I can make the same point in the new design.

Reply:
okay. I dropped that part.